server/services/withings/lib/withings.poll.js

```diff
@@ -0,0 +1,218 @@
+                withingsType = 9;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SYSTOLIC_BLOOD_PRESSURE:
+                withingsType = 10;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.HEARTH_PULSE:
+                withingsType = 11;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.TEMPERATURE:
+                withingsType = 12;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SPO2:
+                withingsType = 54;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.BODY_TEMPERATURE:
+                withingsType = 71;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.SKIN_TEMPERATURE:
+                withingsType = 73;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.MUSCLE_MASS:
+                withingsType = 76;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.HYDRATION:
+                withingsType = 77;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.BONE_MASS:
+                withingsType = 88;
+                break;
+              case DEVICE_FEATURE_TYPES.HEALTH.PULSE_WAVE_VELOCITY:
+                withingsType = 91;
+                break;
+              case DEVICE_FEATURE_TYPES.SENSOR.INTEGER:
+                withingsType = -1;
+                featureBattery = feat;
+                break;
+              default:
+                withingsType = 0;
+                break;
+            }
+
+            if (withingsType > 0) {
+              logger.debug('Current feature last value changed: ', feat.last_value_changed);
+
+              const measureResult = await oauth2Manager.executeQuery(
+                serviceId,
+                user.id,
+                this.integrationName,
+                'get',
+                `${this.withingsUrl}/measure`,
+                `action=getmeas&meastype=${withingsType}&category=1&lastupdate=${feat.last_value_changed.getTime() /
+                  1000 +
+                  1}`,
+              );
+
+              // logger.trace('Poll result : ', measureResult);
+              if (measureResult.data.body.measuregrps) {
+                const mapOfMeasuresGrpsByWithingsDeviceId = new Map();
+                await measureResult.data.body.measuregrps.forEach((element) => {
+                  if (element) {
+                    // Build map of measuregrps by withings device id
+                    let measureList = mapOfMeasuresGrpsByWithingsDeviceId.get(element.deviceid);
+                    if (!measureList) {
+                      measureList = [];
+                    }
+                    measureList.push(element);
+                    mapOfMeasuresGrpsByWithingsDeviceId.set(element.deviceid, measureList);
+                  }
+                });
+
+                await mapOfMeasuresGrpsByWithingsDeviceId.forEach(function buildFeatureByGrps(value, key) {
```

Comment:
Could it be replaced by an arrow function?

Reply:
i transform to arrow function